Add TextBuddy.app v1

ramsey opened the pull request:
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.

miccal commented:
Thank you for your contribution to `homebrew-cask` @ramsey.

Could you please try to add a [`zap` stanza](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md), thanks.

ramsey commented:
Thanks for the feedback, @miccal!

I've added the `zap` stanza. I've also updated the download URL after receiving some feedback from the maintainer: https://twitter.com/TextBuddyApp/status/1362554301219020800

miccal commented:
Thank you @ramsey.

ramsey commented:
Thanks!

Since the download URL always references the latest version of v1, should the version stanza be `1.0.2`? I originally had it as just `1`, but maybe that was the wrong way to do it.

miccal commented:
The version of the `.app` bundle is `1.0.2` as reported by the `livecheck` block, so that is what we use.

ramsey commented:
Thanks for helping and merging so quickly!
